staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -384,6 +403,258 @@ func TestRoundTripAndNewConnection(t *testing.T) {
+					conf = &socks5.Config{
+						AuthMethods: []socks5.Authenticator{authenticator},
+						Rewriter:    interceptor,
+					}
+				} else {
+					conf = &socks5.Config{Rewriter: interceptor}
+				}
+
+				ts, err := socks5.New(conf)
+				if err != nil {
+					t.Errorf("failed to create sock5 server: %v", err)
+				}
+				return ts
+			}
+
+			testCases := map[string]struct {
+				proxyAuth              *url.Userinfo
+				serverConnectionHeader string
+				serverUpgradeHeader    string
+				serverStatusCode       int
+				shouldError            bool
+			}{
+				"proxied tcp->tcp via SOCKS5": {
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+				"proxied tcp->tcp via SOCKS5 with invalid auth": {
+					proxyAuth:              url.UserPassword("invalid", "auth"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            true,
+				},
+				"proxied tcp->tcp via SOCKS5 with auth": {
+					proxyAuth:              url.UserPassword("proxyuser", "proxypasswd"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+			}
+
+			for _, k := range sets.StringKeySet(testCases).List() {
+				testCase := testCases[k]
+				t.Run(k, func(t *testing.T) {
+					server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+						if testCase.shouldError {
+							if e, a := httpstream.HeaderUpgrade, req.Header.Get(httpstream.HeaderConnection); e != a {
+								t.Fatalf("%s: Expected connection=upgrade header, got '%s", k, a)
+							}
+
+							w.Header().Set(httpstream.HeaderConnection, testCase.serverConnectionHeader)
+							w.Header().Set(httpstream.HeaderUpgrade, "SPDY/3.1")
+							w.WriteHeader(testCase.serverStatusCode)
+
+							return
+						}
+
+						streamCh := make(chan httpstream.Stream)
+
+						responseUpgrader := NewResponseUpgrader()
+						spdyConn := responseUpgrader.UpgradeResponse(w, req, func(s httpstream.Stream, replySent <-chan struct{}) error {
+							streamCh <- s
+							return nil
+						})
+						if spdyConn == nil {
+							t.Fatalf("%s: unexpected nil spdyConn", k)
+						}
+						defer spdyConn.Close()
+
+						stream := <-streamCh
+						io.Copy(stream, stream)
+					}))
+					defer server.Close()
+
+					serverURL, err := url.Parse(server.URL)
+					if err != nil {
+						t.Fatalf("%s: Error creating request: %s", k, err)
+					}
+					req, err := http.NewRequest("GET", server.URL, nil)
+					if err != nil {
+						t.Fatalf("%s: Error creating request: %s", k, err)
+					}
+
+					spdyTransport := NewRoundTripper(nil, redirect, redirect)
+
+					var proxierCalled bool
+					var proxyCalledWithHost string
+					var proxyCalledWithAuth bool
+					var proxyCalledWithAuthHeader string
```

Comment:
proxyCalledWithAuth and proxyCalledWithAuthHeader are not really used in the SOCKS5 tests. Can you double check if I'm right and remove the relevant code?

Reply:
you're right, adapted the test